tsl/src/continuous_aggs/materialize.c

```diff
@@ -660,6 +674,66 @@ scan_take_invalidation_tuple(TupleInfo *ti, void *data)
 	return SCAN_CONTINUE;
 }
 
+static ScanTupleResult
+invalidation_threshold_htid_found(TupleInfo *tinfo, void *data)
+{
+	if (tinfo->lockresult != TM_Ok)
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+				 errmsg("unexpected lockresult %d in invalidation_threshold_htid_found",
+						tinfo->lockresult)));
+	}
+	return SCAN_DONE;
+}
+
+/* lock row corresponding to hypertable id in
+ * continuous_aggs_invalidation_threshold table in AccessExclusive mode
+ * block till lock is acquired.
+ */
+static void
+lock_invalidation_threshold_hypertable_row(int32 raw_hypertable_id)
+{
+	ScanTupLock scantuplock = {
+		.waitpolicy = LockWaitBlock,
+		.lockmode = LockTupleExclusive,
+	};
+	Relation invalidation_threshold_table_relation;
+	Catalog *catalog = ts_catalog_get();
+	ScanKeyData scankey[1];
+
+	ScanKeyInit(&scankey[0],
+				Anum_continuous_aggs_invalidation_threshold_pkey_hypertable_id,
+				BTEqualStrategyNumber,
+				F_INT4EQ,
+				Int32GetDatum(raw_hypertable_id));
+
+	/* lock table in RowExclusive mode and the row with AccessExclusive */
+	ScannerCtx scanctx = { .table = catalog_get_table_id(catalog,
```

Comment:
mixed declaration and code. I suggest you move this above the `ScanKeyInit`. It should be safe to initialize the scankey after this declaration even if the scankey is referenced by the scanctx.